src/main/groovy/io/saagie/plugin/dataops/tasks/service/TechnologyService.groovy

```diff
@@ -11,200 +11,212 @@ import org.gradle.api.logging.Logging
+			client.newCall(technologiesVersionRequestCall).execute().withCloseable { response ->
+				handleErrors(response)
+				String responseBody = response.body().string()
+				logger.debug("getTechnologyVersions response $responseBody")
+				def parsedTechnologyVersionsData = slurper.parseText(responseBody)
+				if (!parsedTechnologyVersionsData.data || !parsedTechnologyVersionsData.data.technologiesVersions) {
+					return null
+				}
+				technologiesVersions[technologyId] = parsedTechnologyVersionsData.data.technologiesVersions
+			}
+		}
+		return technologiesVersions[technologyId]
+	}
+	
+	def checkReady() {
+		if (!client || !technologiesRequest || !technologiesVersionsRequest) {
+			throwAndLogError("You need to initialize the singleton TechnologyService with init(...)")
+		}
+	}
+	
+	def checkTechnologiesReady() {
+		if (!technologies) {
+			throwAndLogError("You need to initialize the technologies list")
+		}
+	}
+	
+	def checkTechnologyVersionsReady() {
+		if (!technologiesVersions) {
+			throwAndLogError("You need to initialize the technology versions list")
+		}
+	}
+	
+	static throwAndLogError( message ) {
+		logger.error(message)
+		throw new GradleException(message)
+	}
+	
+	static handleErrors( Response response ) {
+		SaagieUtils.handleErrorClosure(logger, response)
+	}
+	
+	def getV2TechnologyByV1Name( String name ) {
+		checkReady()
+		if (name.equals('java-scala') || name.equals('java')) {
+			return findTechnologyByName('Java/Scala')
+		}
+		
+		if (name.equals('docker')) {
+			return findTechnologyByName('Generic')
+		}
+		
+		findTechnologyByName(name)
+	}
+	
+	def findTechnologyByName( String name ) {
+		checkReady()
+		if (!technologies) {
+			getTechnologies()
+		}
+		checkTechnologiesReady()
+		technologies.find {
+			it.label.toUpperCase().equals(name.toUpperCase())
+		}
+	}
+	
+	def getMostRelevantTechnologyVersion( String technologyId, String versionV1, Map extraTechnologyV1, resultTechnologiesVersions ) {
+		if (!technologiesVersions) {
+			getTechnologiesVersions()
+		}
+		checkTechnologyVersionsReady()
+		
+		def versionV2 = null
+		if (!versionV1) {
+			if (resultTechnologiesVersions != null && resultTechnologiesVersions.first() && resultTechnologiesVersions.first().technologyLabel == "Generic") {
+				versionV2 = resultTechnologiesVersions.first()
+				return [version2 : versionV2, extraTechnology : null]
+			} else {
+				def convertedIntegerList = technologiesVersions[technologyId].collect { version ->
+					return convertToNumber(version.versionLabel)
+				}
+				def index = convertedIntegerList.findIndexOf {
+					it == convertedIntegerList.max()
+				}
+				if (index || index == 0) {
+					versionV2 = technologiesVersions[technologyId][index]
+				}
+			}
+			
+		}
+		
+		if (versionV1) {
+			def upperCaseVersion2 = versionV1.toUpperCase()
+			versionV2 = getTechnologyVersionUntilFind(technologyId, 'versionLabel', upperCaseVersion2)
```

Comment:
can you rename it to be more clear ? i didn't understand `untilFind` what or by what ?